### PR TITLE
Add XDS example to the suite of examples.

### PR DESCRIPTION
**Pull request by donnadionne**
@donnadionne

### CODE REVIEW

**Comment by markdroth** [comment]
I thought the only difference between this example and our existing example is that this one provides a way to specify the target on the command line?  If so, let's just add that functionality to the existing example rather than creating a new one.

File: examples/cpp/helloworld/README.md [hunk 1]
```diff
@@ -255,6 +255,18 @@ main loop in `HandleRpcs` to query the queue.
 
 For a working example, refer to [greeter_async_server.cc](greeter_async_server.cc).
 
+#### XDS client and server
```
Comment (markdroth) [inline examples/cpp/helloworld/README.md]: I don't think we actually need to document anything xds-specific here.  The only thing we're adding here is the `--target` flag, and that could be used to set the target to any URI, even one that has a scheme other than "xds".

All we really need to document is the fact that the `--target` flag can be used if the user wants to point the client at a target other than the default one.

File: examples/cpp/helloworld/greeter_client.cc [hunk 2]
```diff
@@ -73,11 +73,32 @@ class GreeterClient {
...
   // Instantiate the client. It requires a channel, out of which the actual RPCs
-  // are created. This channel models a connection to an endpoint (in this case,
-  // localhost at port 50051). We indicate that the channel isn't authenticated
-  // (use of InsecureChannelCredentials()).
+  // are created. This channel models a connection to an endpoint specified by
+  // the argument "--target=" which is the only expected argument.
+  // We indicate that the channel isn't authenticated (use of
+  // InsecureChannelCredentials()).
+  std::string target_str;
+  std::string arg_str("--target");
```
Comment (markdroth) [inline examples/cpp/helloworld/greeter_client.cc]: How hard would it be to use the gflags library in this example?  That might be easier than writing this argument-parsing code by hand.

**Comment by pgrosu** [comment]
@donnadionne Since you're comfortable with DrRacket/Lisp, just bypass Bazel and use Google Configuration Language (GCL) to simplify things.

**Pull request merged**